sos/report/plugins/ceph_mds.py

```diff
@@ -45,4 +45,60 @@ def setup(self):
             "/etc/ceph/*bindpass*"
         ])
 
+        ceph_cmds = [
+            "cache status",
+            "client ls",
+            "config diff",
+            "config show",
+            "damage ls",
+            "dirfrag ls",
+            "dump cache",
+            "dump inode",
+            "dump loads",
+            "dump snaps",
+            "dump tree",
+            "dump_blocked_ops",
+            "dump_historic_ops",
+            "dump_historic_ops_by_duration",
+            "dump_mempools",
+            "dump_ops_in_flight",
+            "get subtrees",
+            "log dump",
+            "objecter_requests",
+            "openfiles ls",
+            "ops",
+            "perf histogram dump",
+            "perf histogram schema",
+            "perf schema",
+            "perf dump",
+            "status",
+            "version",
+            "session ls"
+        ]
+
+        mds_ids = []
+        # Get the ceph user processes
+        out = self.exec_cmd('ps -u ceph -o args')
+
+        if out['status'] == 0:
+            # Extract the OSD ids from valid output lines
+            for procs in out['output'].splitlines():
+                proc = procs.split()
+                if len(proc) < 6:
+                    continue
+                if proc[4] == '--id' and "ceph-mds" in proc[0]:
+                    mds_ids.append("mds.%s" % proc[5])
+
+        # If containerized, run commands in containers
+        try:
+            cname = self.get_all_containers_by_regex("ceph-mds*")[0][1]
+        except Exception:
+            cname = None
+
+        self.add_cmd_output([
+            "ceph daemon %s %s"
+            % (mdsid, cmd) for mdsid in mds_ids for cmd in ceph_cmds
+        ], container=cname)
```

Comment:
This looks fine by itself, however how many mds ids would a "normal" environment potentially have here?

If there's a decent number of them, we may want to put these collections in an id-specific subdir within the plugin directory, for better organization.

Reply:
I would suppose there would not be more than one, in most cases. I will check with @hillpd to confirm.

Reply:
@hillpd would there ever be multiple mds ids on a single node?

Reply:
Normal environments will not have more than one MDS on a single node.